### ISSUE TITLE
Performance improvements

### ISSUE THREAD

**Issue by aawsome** [green-coder/cdc#2]
I used this crate to chunk a huge amount of data and encountered worse perfomance compared to https://github.com/restic/chunker (this is a Go implementation).

When profiling the results, I see some hot paths within this crate, namely

1) in rolling_hash.rs::slide
```
 let mod_index = (self.hash >> self.polynom_shift) & 255;
```
=> maybe this is not optimized by the compiler as no bound of polynom_shift is known. However polynom_shift is always smaller than 64.
A trick like in https://github.com/restic/chunker/blob/master/chunker.go#L230 might help to get more optimized code.

2) polynom.rs::degree
Seems like the `for` loop is very inefficient here. The Go implementation uses
https://github.com/restic/chunker/blob/master/chunker.go#L230

I would suggest to change this into 
```
    fn degree(&self) -> i32 {
        num_bits::<Self>() as i32 - self.leading_zeros() as i32 - 1
        }
```
or maybe even change the result type to u8 (which should be sufficient here).

**Comment by green-coder** [comment]
Please provide the specs of the CPU on which you run your tests. Was it 0x86 based?

I cannot give a time frame for the fix because maintaining this library is no longer on my priority list.

Also worth mentioning, there are alternative Rust libraries with an emphasis on performance:
- https://github.com/dpc/rdedup
- https://github.com/nlfiedler/fastcdc-rs
- https://github.com/jrobhoward/quickcdc

**Comment by green-coder** [comment]
> Thanks - I already knew about some of those. However, I'm currently working on a rust client for a restic backup repository (https://github.com/restic/restic) and there the CDC is given to use Rabin fingerprints, hence I was looking for such a crate and was very thankful finding yours ;-)

I am glad it helps :-)

A PR would be very welcome, as over the course of the last 5 years my Rust skills ... kind of rusted.

**Comment by green-coder** [comment]
I published the version `0.1.1` of the crate with the improvements. Again, thanks for your contribution.
